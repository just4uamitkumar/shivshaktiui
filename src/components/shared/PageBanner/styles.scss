@import "../../../styles/variables.scss";
@import "../../../styles/colors.scss";
@import "../../../styles/mixins.scss";

.pageBanner {
  background: linear-gradient(90deg, #ff7e5f 0%, #feb47b 100%);
  padding: $grid-4x 0;
  .pageTitle {
    color: $white;
    font-size: 1.8rem;
    font-family: $SodoSansBold;
    line-height: 1.2;
  }
}


