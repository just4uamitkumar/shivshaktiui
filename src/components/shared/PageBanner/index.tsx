import "./styles.scss";
import { Stack, Grid } from "@mui/material";
import CustomBtn from "../../common/Button";
import SkipPrevious from "@mui/icons-material/SkipPrevious";

interface Props {
  title?: string;
  isleftSection?: boolean;
  handleBack?: () => void;
}

const PageBanner: React.FC<Props> = ({ title, isleftSection, handleBack }) => {
  return (
    <>
      <Grid className="pageBanner" container>
        {isleftSection ? (
          <Grid
            className="container"
            justifyContent={"space-between"}
            display={"flex"}
          >
            <Stack>
              <CustomBtn
                btnClass="primary-btn"
                startIcon={<SkipPrevious />}
                text="Back"
                onClick={handleBack}
              />
            </Stack>

            <Stack className="pageTitle">{title}</Stack>
          </Grid>
        ) : (
          <Grid className="container">
            <Stack className="pageTitle">{title}</Stack>
          </Grid>
        )}
      </Grid>
    </>
  );
};

export default PageBanner;
