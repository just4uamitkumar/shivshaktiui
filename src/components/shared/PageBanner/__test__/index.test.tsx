import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import PageBanner from "../index";

describe("PageBanner Component", () => {
  it("matches snapshot", () => {
    const { container } = render(<PageBanner />);
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    render(<PageBanner />);
    expect(PageBanner).toBeDefined();
  });
  it("should render the page title", () => {
    const title = "Test Page";
    render(<PageBanner title={title} />);
    const pageTitle = screen.getByText(title);
    expect(pageTitle).toBeInTheDocument();
  });
  it("should render the back button with correct text", () => {
    const handleBack = vi.fn();
    const title = "Test Page";

    render(
      <PageBanner isleftSection={true} handleBack={handleBack} title={title} />
    );
    const backButton = screen.getByRole("button", { name: /back/i });
    expect(backButton).toBeInTheDocument();
    expect(backButton).toHaveTextContent("Back");
  });
  it("should not render the back button when isleftSection is false", () => {
    const handleBack = vi.fn();

    render(<PageBanner isleftSection={false} handleBack={handleBack} />);
    const backButton = screen.queryByRole("button", { name: /back/i });
    expect(backButton).not.toBeInTheDocument();
  });
  it("should render the back button when isleftSection is true", () => {
    const handleBack = vi.fn();

    render(<PageBanner isleftSection={true} handleBack={handleBack} />);
    const backButton = screen.getByRole("button", { name: /back/i });
    expect(backButton).toBeInTheDocument();
    backButton.click();
    expect(handleBack).toHaveBeenCalled();
  });
});
