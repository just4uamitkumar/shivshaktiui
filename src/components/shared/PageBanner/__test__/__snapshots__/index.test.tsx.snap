// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageBanner Component > matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row pageBanner css-d64g0g-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row container css-17fpwt7-MuiGrid-root"
    >
      <div
        class="MuiStack-root pageTitle css-nen11g-MuiStack-root"
      />
    </div>
  </div>
</div>
`;
